src/wrapper/jnienv.rs

```diff
@@ -2062,6 +2063,95 @@ impl<'a> JNIEnv<'a> {
+    /// returned array will not necessarily be reflected in the original array until
+    /// release_long_array_elements() is called.
+    ///
+    /// See also [`release_long_array_elements`](struct.JNIEnv.html#method.release_long_array_elements)
+    pub fn get_long_array_elements(&self, array: jlongArray) -> Result<(*mut jlong, bool)> {
+        non_null!(array, "get_long_array_elements array argument");
+        let mut is_copy: jboolean = 0xff;
+        let ptr = jni_non_void_call!(self.internal, GetLongArrayElements, array, &mut is_copy);
+        Ok((ptr, is_copy == sys::JNI_TRUE))
+    }
+
+    /// Release elements of the given long array.
+    ///
+    /// Informs the VM that the native code no longer needs access to elems. The elems argument
+    /// is a pointer derived from array using the corresponding get_long_array_elements() function.
+    /// If necessary, this function copies back all changes made to elems to the original array.
+    ///
+    /// The mode argument provides information on how the array buffer should be released. mode
+    /// has no effect if elems is not a copy of the elements in array. Otherwise, mode has the
+    /// following impact:
+    /// CopyBack: Copy back the content and free the elems buffer.
+    /// NoCopyBack: Free the buffer without copying back the possible changes
+    /// Use commit_long_array_elements to copy the elements without releasing the buffer
+    /// (i.e. "commit mode").
+    ///
+    /// In most cases, programmers pass “CopyBack” to the mode argument, to ensure consistent
+    /// behavior for both pinned and copied arrays. The other option gives the programmer more
+    /// control over memory management, and should be used with extreme care.
+    ///
+    /// See also [`commit_long_array_elements`](struct.JNIEnv.html#method.commit_long_array_elements)
+    pub fn release_long_array_elements(
+        &self,
+        array: jlongArray,
+        elems: &mut jlong,
+        mode: ReleaseMode,
+    ) -> Result<()> {
+        non_null!(array, "release_byte_array_elements array argument");
+        jni_void_call!(
+            self.internal,
+            ReleaseLongArrayElements,
+            array,
+            elems,
+            mode as i32
+        );
+        Ok(())
+    }
+
+    /// Commit elements of the given long array.
+    ///
+    /// This function has no effect if elems is not a copy of the elements in array. Otherwise,
+    /// this function copies back the content of the array (and does not free the elems buffer).
+    pub fn commit_long_array_elements(&self, array: jlongArray, elems: &mut jlong) -> Result<()> {
+        non_null!(array, "commit_long_array_elements array argument");
+        jni_void_call!(
+            self.internal,
+            ReleaseLongArrayElements,
+            array,
+            elems,
+            sys::JNI_COMMIT
+        );
+        Ok(())
+    }
+
+    /// Return an AutoLongArray of the given Java long array.
```

Comment:
Could you add please that if an array is `null`, an Error::NullPtr is returned (here and in other auto-arrays)?